Mage/src/main/java/mage/MageObject.java

```diff
@@ -173,9 +180,36 @@ default void addSubType(Game game, SubType... subTypes) {
         }
     }
 
+    default void copySubType(MageObject mageObject, Game game) {
+        copySubType(mageObject, game, null);
+    }
+
+    default void copySubType(MageObject mageObject, Game game, SubTypeSet subTypeSet) {
+        if (subTypeSet == SubTypeSet.CreatureType || subTypeSet == null) {
+            this.setIsAllCreatureTypes(mageObject.isAllCreatureTypes(game), game);
+        }
+        for (SubType subType : getSubtype(game)) {
+            if (subType.getSubTypeSet() == subTypeSet || subTypeSet == null) {
+                continue;
+            }
+            this.addSubType(game, subType);
+        }
+    }
+
     default void removeAllSubTypes(Game game) {
-        getSubtype(game).clear();
-        setIsAllCreatureTypes(false, game);
+        removeAllSubTypes(game, null);
+    }
+
+    default void removeAllSubTypes(Game game, SubTypeSet subTypeSet) {
+        if (subTypeSet == SubTypeSet.CreatureType) {
+            removeAllCreatureTypes(game);
+            return;
+        }
+        if (subTypeSet == SubTypeSet.NonBasicLandType) {
+            getSubtype(game).removeAll(SubType.getLandTypes());
+            return;
+        }
+        getSubtype(game).removeAll(SubType.getBySubTypeSet(subTypeSet));
```

Comment:
Same here -- no removes on null subTypeSet? E.g. `removeAllSubTypes(Game game)` do nothing.

Reply:
this is changed in a later commit